atlasdb-client/src/main/java/com/palantir/atlasdb/logging/KeyValueServiceArgSupplier.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.logging;
+
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.table.description.NameComponentDescription;
+import com.palantir.atlasdb.table.description.NamedColumnDescription;
+import com.palantir.logsafe.Arg;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.UnsafeArg;
+
+public class KeyValueServiceArgSupplier {
+    public static final KeyValueServiceArgSupplier NO_OP = new KeyValueServiceArgSupplier();
+
+    private final KeyValueServiceLogArbitrator logArbitrator;
+
+    public KeyValueServiceArgSupplier(SafeLoggableData safeLoggableData) {
+        this.logArbitrator = new SimpleKeyValueServiceLogArbitrator(safeLoggableData);
+    }
+
+    private KeyValueServiceArgSupplier() {
+        // By default, nothing is safe.
+        this.logArbitrator = new KeyValueServiceLogArbitrator() {
+            @Override
+            public boolean isTableReferenceSafe(TableReference tableReference) {
+                return false;
+            }
+
+            @Override
+            public boolean isRowComponentNameSafe(TableReference tableReference,
+                    NameComponentDescription nameComponentDescription) {
+                return false;
+            }
+
+            @Override
+            public boolean isColumnNameSafe(TableReference tableReference,
+                    NamedColumnDescription namedColumnDescription) {
+                return false;
+            }
+        };
+    }
+
+    public <T> Arg<T> getArgDependingOnTableReference(String name, TableReference tableReference, T value) {
```

Comment:
I wrote this because there were a couple of cases we logged the table reference directly and others where we logged `TableReference.getQualifiedName()`. However, `TableReference`'s toString is `getQualifiedName()`, so I think it's safe.

For the row and column names these aren't used very much yet, so I think it's safe to have them log `getComponentName()` and `getLongName()` respectively. This simplifies the API a fair bit, and we can always add the missing expressivity later if needed.